index.html

```diff
@@ -1492,6 +1492,62 @@ <h3>Types</h3>
 
       </section>
 
+      <section>
+        <h3>Names and Descriptions</h3>
+
+        <p>
+When displaying a <a>credential</a>, it can be useful to have
+text provided by the <a>issuer</a> that provides the
+<a>credential</a> with a name as well as a short description of the
+purpose of the <a>credential</a>. The `name` and `description` <a>properties</a>
+are provided for these purposes.
+        </p>
+
+        <dl>
+          <dt><dfn class="lint-ignore">name</dfn></dt>
+          <dd>
+An OPTIONAL property that expresses the name of the <a>credential</a>. If
+present, the value of the <code>name</code> <a>property</a> MUST be a string.
+Ideally, the name of a <a>credential</a> is concise, human-readable, and could
+be used to enable an individual to quickly differentiate the
+<a>credential</a> from other <a>credentials</a> that they might hold.
+          </dd>
+          <dt><dfn class="lint-ignore">description</dfn></dt>
+          <dd>
+An OPTIONAL property that conveys specific details about a <a>credential</a>.
+If present, the value of the <code>description</code> <a>property</a> MUST be a
+string. Ideally, the description of a <a>credential</a> is no longer than a
+few sentences in length and conveys enough information about the
+<a>credential</a> to remind an individual as to its contents without having
+to look through the entirety of the <a>claims</a>.
+          </dd>
+        </dl>
+
+        <pre class="example nohighlight vc"
+          title="Usage of the name and description property"
+          data-vc-vm="https://university.example/issuers/565049#key-1">
+{
+  "@context": [
```

Comment:
Add `"@language": "en"` ?

Reply:
Fixed in 390360d0f3a2cf5f8f3d4510e8e96f5d8650ca51.

Reply:
@aphillips,

The targeted approach that is currently shown in this PR is a better practice for security and simpler processing than the above (where a language is set context-wide). This is because the term definitions can be protected such that consuming the data without performing JSON-LD transformations would produce the same results as doing so. So both simple consumption and more complex consumption cases will have the same results.

Much of the VCDM is designed around expressing in simple JSON terms what JSON-LD already does, so that simpler processors can just code against the VCDM rules, consuming a subset of JSON-LD syntax without needing additional knowledge or transformations (and this also avoids having to reinvent the wheel that JSON-LD already provides).

Reply:
I don't understand. I thought `@language` in the `@context` was provided by JSON-LD and that, when one has a document in which all of the natural language text fields (such as `name`  and `description`) happen to be in the same language, it would be simpler to declare it in the `@context`. This is likely to be the high-runner case, I suspect, for most credentials. Does it impact the security and processing to show it in this example?

Reply:
@aphillips,

Not everyone wants to have to consume the value of `@language` in `@context` -- so there's better interop if it isn't done that way, but is instead done in a more targeted fashion. Additionally, a number of fields that hold string values that are not natural language text fields would have the `@language` value applied to them, which would create mistakes, for example, base-encoded status lists, any JOSE / SD-JWT fields that are expected to hold string identifiers or base-encoded values, and so on.

Reply:
@aphillips, agree with @dlongley... but I'm going to wait for your confirmation before merging this PR. I think we're doing the most precise, correct thing wrt. i18n for both `name` and `description`... which is all this PR is focusing on. Doing blanket `@language` tags would be problematic for VCs, which we'll probably want to provide guidance on in the i18n section. I've raised issue #1264 to make sure we add this text.

Given that, @aphillips -- what are your thoughts on the above and this PR -- ok to merge, or needs more work?

Reply:
I thought that the `@context`/`@language` mechanism was a common pattern and a decent compromised between no language/direction metadata at all and having it on every natural language string. I think it would be appropriate to specify that the `@language` value does not apply to non-linguistic string data values (just as it doesn't really apply to, say, numbers). Using this mechanism is easy for producers who generate a credential in only a single language and doesn't clutter up the resulting file with lots of repetitive language values. So it's attractive that way. But it doesn't solve the problem of having multiple languages, of course, and it introduces processing that your group (appropriately) cares about. 

I do not have your group's domain expertise here, so would be glad to know if I shouldn't suggest this mechanism.

Other solutions include:

There is `rdf:langString` datatype, which has the downside of not including direction metadata.

JSON-LD 1.1 introduced the [`i18n` namespace](https://www.w3.org/TR/json-ld/#the-i18n-namespace), which allows single-line RDF serialization (which you may not care about) of a string with both language and direction metadata. This fixes some of the serialization issues. I think this is called an `rdf:compoundLiteral` and your PR is (basically) using it in the object representation of `name` and `description` defined here. The downside of this mechanism is that the consumer has to deserialize all of the compound literals to find the best matching one for the consumer's language preference (because the language tag is a field in the object). And both producers and consumers need to expect these (effectively custom) objects.

To solve the searching/language negotiation feature, there are [language maps](https://www.w3.org/TR/json-ld/#language-maps), but they only work with plain strings (to create `langStrings` via a backdoor). You don't show the use of a language map and I didn't suggest it because it doesn't solve the direction problem.

So what am I saying?

Basically, I think you could merge this. But I expect a lot of users will then ignore language and direction and just emit values that look like around line 1550 in the PR. The `compoundLiteral`-like solution exists, but we'll perpetuate a lack of lang/dir metadata:

```javascript
  /* no language, no direction == no rendering information */
  "issuer": {
    "id": "https://university.example/issuers/565049",
    "name": "Example University",
    "description": "A public university focusing on teaching examples."
  },
```

If the goal is to satisfy the _possibility_ of localization by providing the compoundLiteral-like solution, I think that this PR accomplishes that. I am not sure how the string representation of `name` or `description` get lang/dir metadata, though. What am I missing?

I'll look at #1264 and we can go from there.

Reply:
> I'll look at #1264 and we can go from there.

Ok, I'll start engaging on this issue over there.

> If the goal is to satisfy the possibility of localization by providing the compoundLiteral-like solution, I think that this PR accomplishes that.

Yes, that's all this PR is attempting to do -- it's the one thing I know that we can do safely... more in #1264.

> I am not sure how the string representation of name or description get lang/dir metadata, though. What am I missing?

They don't... at least, not automatically. There is more of an explanation in https://github.com/w3c/vc-data-model/issues/1264#issuecomment-1701830685